### ISSUE TITLE
TABLE bayesdb_loom_rowid_mapping makes inappropriate uniqueness assumptions

### ISSUE THREAD

**Issue by Schaechtle** [probcomp/bayeslite#624]
The following table (find it [here](https://github.com/probcomp/bayeslite/blob/189822afc9c3ea1cafc6eeec4cfb43c2a02182c7/src/backends/loom_backend.py#L99)) makes incorrect assumptions regarding uniqueness:
```
CREATE TABLE bayesdb_loom_rowid_mapping (
    generator_id        INTEGER NOT NULL REFERENCES bayesdb_generator(id),
    table_rowid         INTEGER NOT NULL,
    loom_rowid          INTEGER NOT NULL,
    PRIMARY KEY (generator_id, table_rowid)
    UNIQUE(generator_id, table_rowid),
    UNIQUE(generator_id, loom_rowid),
    UNIQUE(table_rowid, loom_rowid)
);
```
One *can* have 2 populations that both have `generator_id` 1 and `table_rowid` 1, thus the first
`UNIQUE` is incorrect.

Suggested fix: remove uniqueness constraint and update PRIMARY KEY to be:

**Comment by fsaad** [comment]
> One can have 2 populations that both have generator_id 1 and table_rowid 1, thus the first `UNIQUE` is incorrect.

That does not sound right, since the `generator_id` is unique, so it cannot be shared by two populations.

I believe it is the final constraint `UNIQUE(table_rowid, loom_rowid)` that needs to be removed, since it disallows the following (perfectly legal) table:
```
| generator_id  | table_rowid   | loom_rowid  |
| ------------- |:-------------:| -----------:|
| 1             |             1 |           1 |
| 2             |             1 |           1 |
```

**Comment by Schaechtle** [comment]
correct. I misread my own edits to the code. Sorry about that. If applied to the correct uniqueness constraint the fix works.
